### PR TITLE
Allow batch_size > 1

### PR DESCRIPTION
**Pull request by hgaiser**
This PR enables the use of `batch_size > 1`. To do this the generators had to be rewritten and the loss layers slightly adjusted.

### CODE REVIEW

File: keras_retinanet/preprocessing/generator.py [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+        batch_size=1,
+        group_method='ratio', # one of 'none', 'random', 'ratio'
+        shuffle_groups=True,
+        image_min_side=600,
+        image_max_side=1024,
+    ):
+        self.image_data_generator = image_data_generator
+        self.batch_size           = int(batch_size)
+        self.group_method         = group_method
+        self.shuffle_groups       = shuffle_groups
```
[the remark this answers is not in the document]
Comment (de-vri-es) [reply keras_retinanet/preprocessing/generator.py]: the indentation looks good to me

File: examples/train_coco.py [hunk 2]
```diff
@@ -35,13 +35,14 @@ def get_session():
 
 def create_model(weights='imagenet'):
     image = keras.layers.Input((None, None, 3))
-    return ResNet50RetinaNet(image, num_classes=90, weights=weights)
+    return ResNet50RetinaNet(image, num_classes=80, weights=weights)
```
[the remark this answers is not in the document]
Comment (hgaiser) [reply examples/train_coco.py]: COCO actually only has 80 classes, but their max category id is 90 (there are some holes in the order of the category ids).

**Comment by Arthaslixin** [comment]
What's the recommended batch size for a single Titan X card?

**Comment by de-vri-es** [comment]
> What's the recommended batch size for a single Titan X card?

@Arthaslixin: Please use the github issue tracker only for issues with or improvements of this software implementation. Discussions like this are better suited for the slack channel (see the README for details).

**Comment by hgaiser** [comment]
I'm not sure what you're asking exactly? Simply update the repository and you should have this update.
